### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by itdove**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ jobs:
       script:
         - |
           docker build --tag quay.io/open-cluster-management/build-harness:latest .
-          docker login quay.io/open-cluster-management -u ${DOCKER_USER} -p ${DOCKER_PASS}
-          docker push quay.io/open-cluster-management/build-harness:latest
+          if [ "$TRAVIS_REPO_SLUG" == "open-cluster-management/build-harness-extensions" ]; then
```
Comment (arewm) [inline .travis.yml:28]: Can you add an additional condition here for `TRAVIS_EVENT_TYPE ` is not "pull_request". That will prevent it from causing failures in PRs.

Comment (itdove) [reply .travis.yml:28]: k

Comment (itdove) [reply .travis.yml:28]: Done

File: modules/component/Makefile [hunk 2]
```diff
@@ -46,6 +56,20 @@ component/build: %build: %init
 	${COMPONENT_BUILD_COMMAND} ${COMPONENT_DOCKER_REPO}/${COMPONENT_NAME}:${COMPONENT_VERSION}${COMPONENT_TAG_EXTENSION}
 	@echo "Built the following image: ${COMPONENT_DOCKER_REPO}/${COMPONENT_NAME}:${COMPONENT_VERSION}${COMPONENT_TAG_EXTENSION}"
 
+.PHONY: component/build-coverage
+## Build the component
+component/build-coverage: %build-coverage: %init %build
```
Comment (schmidtd) [inline modules/component/Makefile:61]: If you're going to prereq %build, that target already calls (prereqs) %init

File: modules/component/Makefile [hunk 3]
```diff
@@ -20,6 +26,10 @@ IMAGE_DEPLOYED_NAME ?=
 COMPONENT_INIT_COMMAND ?= ${COMPONENT_SCRIPTS_PATH}/install-dependencies.sh
 # The command to run to execute a build
 COMPONENT_BUILD_COMMAND ?= ${COMPONENT_SCRIPTS_PATH}/build.sh
+# The command to run to execute a build the image coverage
+COMPONENT_BUILD_COVERAGE_COMMAND ?= ${COMPONENT_SCRIPTS_PATH}/build-coverage.sh
+# The command to run to execute a build the image e2e
+COMPONENT_BUILD_E2E_COMMAND ?= ${COMPONENT_SCRIPTS_PATH}/build-e2e.sh
```
Comment (schmidtd) [inline modules/component/Makefile:32]: Will you be contributing example `build-coverage.sh` and `build-e2e.sh` commands to demonstrate their calling convention/parameter passing?

File: modules/component/Makefile [hunk 4]
```diff
@@ -96,6 +122,28 @@ component/push: %push:
 	$(DOCKER) push ${COMPONENT_DOCKER_REPO}/${COMPONENT_NAME}:${COMPONENT_VERSION}${COMPONENT_TAG_EXTENSION}
 	@echo "Pushed the following image: ${COMPONENT_DOCKER_REPO}/${COMPONENT_NAME}:${COMPONENT_VERSION}${COMPONENT_TAG_EXTENSION}"
 
+.PHONY: component/push-coverage
+## Push the component to COMPONENT_DOCKER_REPO
```
Comment (schmidtd) [inline modules/component/Makefile:126]: ## Push the component's coverage image to COMPONENT_DOCKER_REPO

File: modules/component/Makefile [hunk 5]
```diff
@@ -96,6 +122,28 @@ component/push: %push:
...
+	$(call assert-set,DOCKER_USER)
+	$(call assert-set,DOCKER_PASS)
+	$(call assert-set,COMPONENT_NAME)
+	$(call assert-set,COMPONENT_VERSION)
+	$(DOCKER) login ${COMPONENT_DOCKER_REPO} -u ${DOCKER_USER} -p ${DOCKER_PASS}
+	$(DOCKER) push ${COMPONENT_DOCKER_REPO}/${COMPONENT_NAME}${DOCKER_IMAGE_COVERAGE_POSTFIX}:${COMPONENT_VERSION}${COMPONENT_TAG_EXTENSION}
+	@echo "Pushed the following image: ${COMPONENT_DOCKER_REPO}/${COMPONENT_NAME}${DOCKER_IMAGE_COVERAGE_POSTFIX}:${COMPONENT_VERSION}${COMPONENT_TAG_EXTENSION}"
+
+.PHONY: component/push-e2e
+## Push the component to COMPONENT_DOCKER_REPO
```
Comment (schmidtd) [inline modules/component/Makefile:137]: `## Push the component's e2e testing image to COMPONENT_DOCKER_REPO`

**Review by schmidtd — CHANGES_REQUESTED** [review]
Looks like the proposed changes should be safe for users to adopt at their own pace.  I'd request stub shell scripts for the ones you define that are similar to the examples already provided (i.e. they don't really do anything but explain what they are supposed to do and `exit 1` as well as the other suggestions above.

File: modules/component/Makefile (hunk 2 above)
Comment (itdove) [reply modules/component/Makefile:61]: ok

File: modules/component/Makefile (hunk 3 above)
Comment (itdove) [reply modules/component/Makefile:32]: where I can post examples?

Comment (schmidtd) [reply modules/component/Makefile:32]: They should go right where you defined them... `component/bin`

**Comment by itdove** [comment]
@dschmidtAs recommended by @joeg-pro I added the https://docs.google.com/document/d/1YTqpZRH54Bnn4WJ2nZmjaCoiRtqmrc2w6DdQxe_yLZ8/edit#heading=h.1ocirhrp51lj section which explain how to generate the coverage image. This contains now also links to code sample.

**Comment by itdove** [comment]
No-op scripts added.

**Review by schmidtd — APPROVED** [review]
:rocket:

**Pull request merged**